src/api/cvc4cpp.cpp

```diff
@@ -2136,6 +2136,25 @@ Term DatatypeConstructor::getConstructorTerm() const
   return ctor;
 }
 
+Term DatatypeConstructor::getSpecializedConstructorTerm(Sort retSort) const
+{
+  CVC4_API_SOLVER_TRY_CATCH_BEGIN;
+  
+  NodeManager* nm = d_solver->getNodeManager();
+  // apply type ascription to the operator
+  Term sctor =
+      api::Term(d_solver,
+                nm->mkNode(kind::APPLY_TYPE_ASCRIPTION,
+                           nm->mkConst(AscriptionType(
+                               d_ctor
+                                   ->getSpecializedConstructorType(
+                                       retSort.getType()))),
+                           d_ctor->getConstructor()));
```

Comment:
Don't you have to kick off type checking for this term?

Reply:
Yes, good catch, I've added this.